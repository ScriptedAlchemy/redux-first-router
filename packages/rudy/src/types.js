// @flow
export const PREFIX = '@@rudy'
export const prefixType = (type: string, prefix?: string) =>
  `${prefix || PREFIX}/${type}`

export const ADD_ROUTES = prefixType('ADD_ROUTES')
