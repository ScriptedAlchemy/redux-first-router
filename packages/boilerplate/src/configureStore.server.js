import { doesRedirect } from '@respond-framework/rudy'
import configureStore from './configureStore.browser'

export default async (req, res) => {
  const { store, firstRoute } = configureStore(undefined, req.path)
  const result = await store.dispatch(firstRoute())
  if (doesRedirect(result, res)) return false

  const { status } = store.getState().location
  res.status(status)

  return store
}
