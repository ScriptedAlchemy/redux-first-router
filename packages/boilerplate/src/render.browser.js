/* eslint-env browser */

import React from 'react'
import ReactDOM from 'react-dom'
import { Provider } from 'react-redux'
import App from './components/App'
import configureStore from './configureStore'

const { store, firstRoute } = configureStore(window.REDUX_STATE)

const root = document.getElementById('root')

const render = () =>
  ReactDOM.hydrate(
    <Provider store={store}>
      <App />
    </Provider>,
    root,
  )

store.dispatch(firstRoute()).then(() => {
  render()
})

if (module.hot) {
  module.hot.accept('./components/App', render)
}
