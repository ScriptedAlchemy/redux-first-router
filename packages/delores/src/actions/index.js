export { default as redirect } from './redirect'
export { default as changeBasename } from './changeBasename'
export { default as clearCache } from './clearCache'
export { default as confirm } from './confirm'

export {
  push,
  replace,
  jump,
  back,
  next,
  reset,
  set,
  setParams,
  setQuery,
  setState,
  setHash,
  setBasename,
} from './history'
