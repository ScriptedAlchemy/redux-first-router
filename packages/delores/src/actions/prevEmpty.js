export const createPrevEmpty = (): Prev => ({
  type: '',
  params: {},
  query: {},
  state: {},
  hash: '',
  basename: '',
  location: {
    url: '',
    pathname: '',
    search: '',
    key: '',
    scene: '',
    index: -1,
  },
})
