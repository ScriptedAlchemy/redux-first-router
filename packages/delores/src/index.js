export * from './utils/supports'
export * from './utils/popListener'
export * from './utils/sessionStorage'

export { default as toEntries, findInitialN } from './utils/toEntries'
export { default as toAction } from './utils/toAction'
export { default as actionToUrl } from './utils/actionToUrl'
export { default as urlToAction, findBasename, stripBasename } from './utils/urlToAction'
export { default as locationToUrl } from './utils/locationToUrl'
export { default as urlToLocation } from './utils/urlToLocation'
export { default as matchUrl } from './utils/matchUrl'
export { default as compileUrl } from './utils/compileUrl'
export { default as cleanBasename } from './utils/cleanBasename'

export { default as notFound } from './actions/notFound'

export { default as History } from './History'
export { default as MemoryHistory } from './MemoryHistory'
export { default as BrowserHistory } from './BrowserHistory'
