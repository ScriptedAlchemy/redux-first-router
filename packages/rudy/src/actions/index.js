export { default as addRoutes } from './addRoutes'
