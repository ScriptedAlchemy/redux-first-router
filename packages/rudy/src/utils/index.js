export { default as isHydrate } from './isHydrate'
export { default as isAction } from './isAction'
export { default as isNotFound } from './isNotFound'
export { default as isServer } from './isServer'
export { default as isRedirect } from './isRedirect'

export { default as doesRedirect } from './doesRedirect'
export { default as shouldTransition } from './shouldTransition'

export { default as formatRoutes, formatRoute } from './formatRoutes'
export { default as typeToScene } from './typeToScene'

export { default as redirectShortcut } from './redirectShortcut'
export { default as callRoute } from './callRoute'

export { default as noOp } from './noOp'

export { default as createSelector } from './createSelector'

export { default as nestAction, createActionRef } from './nestAction'

export { default as logError, onError } from './logError'

export { default as parseSearch } from './parseSearch'