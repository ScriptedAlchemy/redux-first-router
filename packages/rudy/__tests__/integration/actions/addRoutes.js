import createTest from '../../../__helpers__/createTest'
import { addRoutes } from '../../../src/actions'

createTest(
  'dispatch(addRoutes(routes))',
  {},
  [],
  async ({ snap, getState }) => {
    const routes = {
      FOO: {
        path: '/foo',
      },
    }

    await snap(addRoutes(routes))
    await snap({ type: 'FOO' })

    expect(getState().location.pathname).toEqual('/foo')
  },
)

createTest(
  'dispatch(addRoutes(routes, formatRoute))',
  {},
  [],
  async ({ snap, getState, dispatch }) => {
    const routes = {
      FOO: {
        path: '/foo',
        thunk: 'BAZ',
      },
      BAZ: {
        path: '/baz',
        thunk: () => 'payload!',
      },
    }

    const formatRoute = (route) => ({
      ...route,
      path: '/bar',
    })

    await snap(addRoutes(routes, formatRoute))
    await snap({ type: 'FOO' })

    expect(getState().location.pathname).toEqual('/bar')
  },
)
