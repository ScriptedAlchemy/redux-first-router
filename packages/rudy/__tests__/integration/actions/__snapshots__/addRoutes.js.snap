// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dispatch(addRoutes(routes)): {"type":"@@rudy/ADD_ROUTES","payload":{"routes":{"FOO":{"path":"/foo"}}}} - response 1`] = `
Object {
  "payload": Object {
    "formatRoute": undefined,
    "routes": Object {
      "FOO": Object {
        "path": "/foo",
        "type": "FOO",
      },
    },
    "routesAdded": 1,
  },
  "type": "@@rudy/ADD_ROUTES",
}
`;

exports[`dispatch(addRoutes(routes)): {"type":"@@rudy/ADD_ROUTES","payload":{"routes":{"FOO":{"path":"/foo"}}}} - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`dispatch(addRoutes(routes)): {"type":"@@rudy/ADD_ROUTES","payload":{"routes":{"FOO":{"path":"/foo"}}}} - routes - REDIRECTED - onComplete 1`] = `0`;

exports[`dispatch(addRoutes(routes)): {"type":"@@rudy/ADD_ROUTES","payload":{"routes":{"FOO":{"path":"/foo"}}}} - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/first",
          "scene": "",
          "search": "",
          "url": "/first",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
    ],
    "from": null,
    "hash": "",
    "index": 0,
    "key": "345678",
    "kind": "load",
    "length": 1,
    "n": 1,
    "params": Object {},
    "pathname": "/first",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": -1,
        "key": "",
        "pathname": "",
        "scene": "",
        "search": "",
        "url": "",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "",
    },
    "query": Object {},
    "routesAdded": 1,
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 200,
    "type": "FIRST",
    "universal": false,
    "url": "/first",
  },
  "title": "@@rudy/ADD_ROUTES - {\\"routes\\":{\\"FOO\\":{\\"path\\":\\"/foo\\",\\"type\\":\\"FOO\\"}},\\"routesAdded\\":1}",
}
`;

exports[`dispatch(addRoutes(routes)): {"type":"@@rudy/ADD_ROUTES","payload":{"routes":{"FOO":{"path":"/foo"}}}} - title 1`] = `"FIRST"`;

exports[`dispatch(addRoutes(routes)): {"type":"FOO"} - response 1`] = `
Object {
  "basename": "",
  "hash": "",
  "location": Object {
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/first",
          "scene": "",
          "search": "",
          "url": "/first",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/foo",
          "scene": "",
          "search": "",
          "url": "/foo",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FOO",
      },
    ],
    "from": null,
    "index": 1,
    "key": "345678",
    "kind": "push",
    "length": 2,
    "n": 1,
    "pathname": "/foo",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 0,
        "key": "345678",
        "pathname": "/first",
        "routesAdded": 1,
        "scene": "",
        "search": "",
        "url": "/first",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "FIRST",
    },
    "scene": "",
    "search": "",
    "status": 200,
    "url": "/foo",
  },
  "params": Object {},
  "query": Object {},
  "state": Object {},
  "type": "FOO",
}
`;

exports[`dispatch(addRoutes(routes)): {"type":"FOO"} - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`dispatch(addRoutes(routes)): {"type":"FOO"} - routes - REDIRECTED - onComplete 1`] = `0`;

exports[`dispatch(addRoutes(routes)): {"type":"FOO"} - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/first",
          "scene": "",
          "search": "",
          "url": "/first",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/foo",
          "scene": "",
          "search": "",
          "url": "/foo",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FOO",
      },
    ],
    "from": null,
    "hash": "",
    "index": 1,
    "key": "345678",
    "kind": "push",
    "length": 2,
    "n": 1,
    "params": Object {},
    "pathname": "/foo",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 0,
        "key": "345678",
        "pathname": "/first",
        "routesAdded": 1,
        "scene": "",
        "search": "",
        "url": "/first",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "FIRST",
    },
    "query": Object {},
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 200,
    "type": "FOO",
    "universal": false,
    "url": "/foo",
  },
  "title": "FOO",
}
`;

exports[`dispatch(addRoutes(routes)): {"type":"FOO"} - title 1`] = `"FOO"`;

exports[`dispatch(addRoutes(routes, formatRoute)): {"type":"@@rudy/ADD_ROUTES","payload":{"routes":{"FOO":{"path":"/foo","thunk":"BAZ"},"BAZ":{"path":"/baz"}}}} - response 1`] = `
Object {
  "payload": Object {
    "formatRoute": [Function],
    "routes": Object {
      "BAZ": Object {
        "path": "/bar",
        "thunk": [Function],
        "type": "BAZ",
      },
      "FOO": Object {
        "path": "/bar",
        "thunk": [Function],
        "type": "FOO",
      },
    },
    "routesAdded": 2,
  },
  "type": "@@rudy/ADD_ROUTES",
}
`;

exports[`dispatch(addRoutes(routes, formatRoute)): {"type":"@@rudy/ADD_ROUTES","payload":{"routes":{"FOO":{"path":"/foo","thunk":"BAZ"},"BAZ":{"path":"/baz"}}}} - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`dispatch(addRoutes(routes, formatRoute)): {"type":"@@rudy/ADD_ROUTES","payload":{"routes":{"FOO":{"path":"/foo","thunk":"BAZ"},"BAZ":{"path":"/baz"}}}} - routes - REDIRECTED - onComplete 1`] = `0`;

exports[`dispatch(addRoutes(routes, formatRoute)): {"type":"@@rudy/ADD_ROUTES","payload":{"routes":{"FOO":{"path":"/foo","thunk":"BAZ"},"BAZ":{"path":"/baz"}}}} - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/first",
          "scene": "",
          "search": "",
          "url": "/first",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
    ],
    "from": null,
    "hash": "",
    "index": 0,
    "key": "345678",
    "kind": "load",
    "length": 1,
    "n": 1,
    "params": Object {},
    "pathname": "/first",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": -1,
        "key": "",
        "pathname": "",
        "scene": "",
        "search": "",
        "url": "",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "",
    },
    "query": Object {},
    "routesAdded": 2,
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 200,
    "type": "FIRST",
    "universal": false,
    "url": "/first",
  },
  "title": "@@rudy/ADD_ROUTES - {\\"routes\\":{\\"FOO\\":{\\"path\\":\\"/bar\\",\\"type\\":\\"FOO\\"},\\"BAZ\\":{\\"path\\":\\"/bar\\",\\"type\\":\\"BAZ\\"}},\\"routesAdded\\":2}",
}
`;

exports[`dispatch(addRoutes(routes, formatRoute)): {"type":"@@rudy/ADD_ROUTES","payload":{"routes":{"FOO":{"path":"/foo","thunk":"BAZ"},"BAZ":{"path":"/baz"}}}} - title 1`] = `"FIRST"`;

exports[`dispatch(addRoutes(routes, formatRoute)): {"type":"FOO"} - response 1`] = `
Object {
  "payload": "payload!",
  "type": "FOO_COMPLETE",
}
`;

exports[`dispatch(addRoutes(routes, formatRoute)): {"type":"FOO"} - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`dispatch(addRoutes(routes, formatRoute)): {"type":"FOO"} - routes - REDIRECTED - onComplete 1`] = `0`;

exports[`dispatch(addRoutes(routes, formatRoute)): {"type":"FOO"} - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/first",
          "scene": "",
          "search": "",
          "url": "/first",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/bar",
          "scene": "",
          "search": "",
          "url": "/bar",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FOO",
      },
    ],
    "from": null,
    "hash": "",
    "index": 1,
    "key": "345678",
    "kind": "push",
    "length": 2,
    "n": 1,
    "params": Object {},
    "pathname": "/bar",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 0,
        "key": "345678",
        "pathname": "/first",
        "routesAdded": 2,
        "scene": "",
        "search": "",
        "url": "/first",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "FIRST",
    },
    "query": Object {},
    "ready": true,
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 200,
    "type": "FOO",
    "universal": false,
    "url": "/bar",
  },
  "title": "FOO_COMPLETE - \\"payload!\\"",
}
`;

exports[`dispatch(addRoutes(routes, formatRoute)): {"type":"FOO"} - title 1`] = `"FOO"`;
